ast.c

```diff
@@ -1518,6 +1522,10 @@ PHP_MINIT_FUNCTION(ast) {
 	ast_declare_property(ast_node_ce, AST_STR(str_flags), &zv_null);
 	ast_declare_property(ast_node_ce, AST_STR(str_lineno), &zv_null);
 	ast_declare_property(ast_node_ce, AST_STR(str_children), &zv_null);
+#ifdef ZEND_ACC_ALLOW_DYNAMIC_PROPERTIES
+	zend_add_class_attribute(ast_node_ce, zend_ce_allow_dynamic_properties->name, 0);
+	ast_node_ce->ce_flags |= ZEND_ACC_ALLOW_DYNAMIC_PROPERTIES;
```

Comment:
Setting the flag explicitly should not be necessary.

Reply:
I tried removing the line. `Deprecated: Creation of dynamic property ast\Node::$undeclaredDynamic is deprecated in /home/tyson/programming/php-ast/tests/php82_metadata.php on line 18` is the result.


This is the same as what https://github.com/php/php-src/blob/master/Zend/zend_builtin_functions.c does

Reply:
Ah yes, looks like the callback doesn't get called on interned classes for some reason.